SoptSeminar/Presentation/FirstViewController.swift

```diff
@@ -0,0 +1,223 @@
+            self.view.frame.origin.y = distance > 0 ? -distance : 0
+        }
+    }
+    
+    @objc private func keyboardWillHide() {
+        UIView.animate(withDuration: 0.25) {
+            self.view.frame.origin.y = 0
+        }
+    }
+
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        view.endEditing(true)
+        super.touchesBegan(touches, with: event)
+    }
+    
+    func presentToSecondViewController() {
+        let secondVC = SecondViewController()
+        secondVC.modalPresentationStyle = .fullScreen
+        secondVC.getUserName(name: name)
+        self.present(secondVC, animated: true)
+    }
+    
+    func pushToSecondViewController() {
+        let secondVC = SecondViewController()
+        secondVC.getUserName(name: name)
+        self.navigationController?.pushViewController(secondVC, animated: true)
+    }
+    
+    @objc
+    func presentButtonTapped() {
+        presentToSecondViewController()
+    }
+    
+    @objc
+    func pushButtonTapped() {
+        pushToSecondViewController()
+    }
+}
+
+// MARK: - UI & Layout
+
+private extension FirstViewController {
+    
+    func setUI() {
+        view.backgroundColor = .white
+        nameTextField.setLeftPaddingPoints(15)
+    }
+    
+    func setLayout() {
+        
+        view.addSubviews(containerView)
+        containerView.addSubviews(nameLabel, nameTextField, errorLabel, presentButton, pushButton)
+        
+        containerView.snp.makeConstraints {
+            $0.top.equalToSuperview().offset(400)
+            $0.leading.trailing.equalToSuperview().inset(30)
+            $0.height.equalTo(250)
+        }
+        
+        nameLabel.snp.makeConstraints {
+            $0.top.equalToSuperview()
+            $0.centerX.equalToSuperview()
+        }
+        
+        nameTextField.snp.makeConstraints {
+            $0.top.equalTo(nameLabel.snp.bottom).offset(20)
+            $0.leading.trailing.equalToSuperview()
+            $0.height.equalTo(48)
+        }
+        
+        errorLabel.snp.makeConstraints {
+            $0.top.equalTo(nameTextField.snp.bottom).offset(5)
+            $0.trailing.equalTo(nameTextField)
+        }
+        
+        presentButton.snp.makeConstraints {
+            $0.top.equalTo(nameTextField.snp.bottom).offset(30)
+            $0.centerX.equalToSuperview()
+            $0.height.equalTo(40)
+        }
+        
+        pushButton.snp.makeConstraints {
+            $0.top.equalTo(presentButton.snp.bottom).offset(20)
+            $0.centerX.equalToSuperview()
+            $0.height.equalTo(48)
+        }
+    }
+    
+    private func setupLayoutForEmptyName() {
+        errorLabel.isHidden = false
+        presentButton.isEnabled = false
+        pushButton.isEnabled = false
+    }
+    
+    private func setupLayoutForNonEmptyName() {
+        errorLabel.isHidden = true
+        presentButton.isEnabled = true
+        pushButton.isEnabled = true
+    }
```

Comment:
이렇게 TextField 값 유무에 따라서 boolean값이 나뉘는 것은 isEmpty속성을 이용해서 좀 더 간결하게 표현할 수 있을 것 같아요 !!

```suggestion
    private func setupLayoutForTextField() {
        if let name = nameTextField.text {
            self.name = name
        }
        errorLabel.isHidden = !(name.isEmpty)
        presentButton.isEnabled = !(name.isEmpty)
        pushButton.isEnabled = !(name.isEmpty)
    }
```